repos/system_upgrade/el7toel8/actors/initramdiskgenerator/libraries/initramgen.py

```diff
@@ -26,92 +34,115 @@ def copy_dracut_modules(context, modules):
-    for message in api.consume(RequiredUpgradeInitramPackages):
-        packages.update(message.packages)
     dnfplugin.install_initramdisk_requirements(
-        packages=packages, target_userspace_info=target_userspace_info, used_repos=used_repos)
+        packages=packages,
+        target_userspace_info=target_userspace_info,
+        used_repos=used_repos)
 
 
-def _install_files(context):
+# duplicate of _copy_files fro userspacegen.py
+def _copy_files(context, files):
     """
-    Copy the required files into the `context` userspace and return the list
-    of those files.
+    Copy the files/dirs from the host to the `context` userspace
 
-    Those files will be copied from the source to the userspace used for
-    generating of the upgrade initrd. These files may need to be installed
-    in the initrd explicitly, so return the list of those files for other
-    purposes.
+    :param context: An instance of a mounting.IsolatedActions class
+    :type context: mounting.IsolatedActions class
+    :param files: list of files that should be copied from the host to the context
+    :type files: list of CopyFile
     """
-    # TODO: currently we need this just for /etc/dasd.conf, but it's expected
-    # we will need this for more files in future. The concept used here will
-    # need to be changed, as we should consume specific messages instead.
-    # But for now this should be enough. Keeping that for future.
-    if architecture.matches_architecture(architecture.ARCH_S390X):
-        # we don't need to check existence of the file - it is required on this
-        # this architecture
-        context.copy_to('/etc/dasd.conf', '/etc/dasd.conf')
-        return ['/etc/dasd.conf']
-    return []
-
-
-def install_multipath_files(context):
-    # Include multipath related files (according to module-setup of multipath)
-    if os.path.exists('/etc/xdrdevices.conf'):
-        context.copy_to('/etc/xdrdevices.conf', '/etc/xdrdevices.conf')
-    if os.path.exists('/etc/multipath.conf'):
-        context.copy_to('/etc/multipath.conf', '/etc/multipath.conf')
-        if os.path.isdir('/etc/multipath'):
-            shutil.rmtree(context.full_path('/etc/multipath'))
-            context.copytree_to('/etc/multipath', '/etc/multipath')
+    for file_task in files:
+        if not file_task.dst:
+            file_task.dst = file_task.src
+        if os.path.isdir(file_task.src):
+            context.remove_tree(file_task.dst)
+            context.copytree_to(file_task.src, file_task.dst)
+        else:
+            context.copy_to(file_task.src, file_task.dst)
+
+
+# TODO: think about possibility to split this part to different actor
+# # reasoning: the environment could be prepared automatically and actor
+# # developers will be able to do additional modifications before the initrd
+# # will be really generated. E.g. multipath: config files will be copied
+# # and another actor can securely updated configuration before the initrd
```

Comment:
What would be the benefit in case of multipath config files? But otherwise I agree...

Reply:
The hardcoded copying will be removed.